File: classes/Collection.php
<?php
 /**
  * Representation of a Zotero Collection
  *
  * @copyright  Copyright (c) 2008  Center for History and New Media (http://chnm.gmu.edu)
  * @license    http://www.opensource.org/licenses/ecl1.php    ECL License
  * @since      Class available since Release 0.0
  * @see        Zotero_Entry
  */
class Zotero_Collection extends Zotero_Entry
{
    /**
     * @var int
     */
    public $collectionKey;
    
    /**
     * @var int
     */
    public $numCollections;
    
    /**
     * @var int
     */
    public $numItems;
    
    /**
     * @var string
     */
    public $parentCollectionKey;
    
    public function __construct($entryNode)
    {
        parent::__construct($entryNode);
        // Extract the collectionKey
        $this->collectionKey = $entryNode->getElementsByTagNameNS('*', 'key')->item(0)->nodeValue;
        $this->numCollections = $entryNode->getElementsByTagName('numCollections')->item(0)->nodeValue;
        $this->numItems = $entryNode->getElementsByTagName('numItems')->item(0)->nodeValue;
        
        $contentNode = $entryNode->getElementsByTagName('content')->item(0);
        $contentType = parent::getContentType($entryNode);
        if($contentType == 'application/json'){
            $this->contentArray = json_decode($contentNode->nodeValue, true);
            $this->etag = $contentNode->getAttribute('etag');
        }
        elseif($contentType == 'xhtml'){
            $this->parseXhtmlContent($contentNode);
        }
        
    }
    
    public function dataObject() {
        $jsonItem = new stdClass;
        
        //inherited from Entry
        $jsonItem->title = $this->title;
        $jsonItem->dateAdded = $this->dateAdded;
        $jsonItem->dateUpdated = $this->dateUpdated;
        $jsonItem->id = $this->id;
        
        $jsonItem->collectionKey = $this->collectionKey;
        foreach($this->entries as $entry){
            $jsonItem->entries[] = $entry->dataObject();
        }
        
        return $jsonItem;
    }
}

File: lib/py/libZotero/collection.py
#import logging
import json
from entry import *


class Collection(Entry):
    def __init__(self, entryNode=None, library=None):
        self.libraryUrlIdentifier = ''
        self.title = ''
        self.name = ''
        self.collectionKey = None
        self.collectionVersion = 0
        self.numCollections = 0
        self.numItems = 0
        self.parentCollectionKey = False
        self.apiObject = {}
        self.pristine = {}

        if library != None:
            self.associateWithLibrary(library)
        if entryNode == None:
            return
        super(Collection, self).__init__(entryNode)
        self.collectionKey = entryNode.getElementsByTagNameNS('http://zotero.org/ns/api', 'key').item(0).childNodes.item(0).nodeValue
        self.collectionVersion = entryNode.getElementsByTagNameNS('http://zotero.org/ns/api', 'version').item(0).childNodes.item(0).nodeValue
        self.numCollections = entryNode.getElementsByTagName('http://zotero.org/ns/api', 'numCollections').item(0).childNodes.item(0).nodeValue
        self.numItems = entryNode.getElementsByTagNameNS('http://zotero.org/ns/api', 'numItems').item(0).childNodes.item(0).nodeValue

        contentNode = entryNode.getElementsByTagName('content').item(0)
        if contentNode:
            contentType = contentNode.getAttribute('type')
            if contentType == 'application/json':
                jsonstring = contentNode.childNodes.item(0).nodeValue
                self.pristine = json.loads(jsonstring)
                self.apiObject = json.loads(jsonstring)
                self.parentCollectionKey = self.apiObject['parentCollection']
                self.name = self.apiObject['name']

    def writeApiObject(self):
        return dict(self.pristine.items() + self.apiObject.items())

    def get(self, key):
        if key == 'title' or key == 'name':
            return self.name
        elif key == 'collectionKey' or key == 'key':
            return self.collectionKey
        elif key == 'collectionVersion' or key == 'version':
            return self.collectionVersion
        elif key == 'parentCollection' or key == 'parentCollectionKey':
            return self.parentCollectionKey

        if key in self.apiObject:
            return self.apiObject[key]

        if key in dir(self):
            return self.__getattribute__(key)

        return None

    def set(self, key, val):
        if key in self.apiObject:
            self.apiObject[key] = val

        if key == 'title' or key == 'name':
            self.name = val
            self.apiObject['name'] = val
        elif key == 'collectionKey' or key == 'key':
            self.collectionKey = val
            self.apiObject['collectionKey'] = val
        elif key == 'parentCollection' or key == 'parentCollectionKey':
            self.parentCollectionKey = val
            self.apiObject['parentCollection'] = val
        elif key == 'collectionVersion' or key == 'version':
            self.collectionVersion = val
            self.apiObject['collectionVersion'] = val

        if key in self.apiObject:
            self.apiObject[key] = val

        if key in dir(self):
            self.__setattr__(key, val)
